extension/CMakeLists.txt

```diff
@@ -3,14 +3,42 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 ## Custom target to build the web-extension bridge with Rust/Cargo
-add_custom_target(cargo_mozillavpnnp ALL
-    COMMENT "Building web extension crate"
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bridge
-    COMMAND ${CMAKE_COMMAND} -E env CARGO_HOME=${CMAKE_CURRENT_BINARY_DIR}/.cargo_home
-            cargo build --release --target-dir ${CMAKE_CURRENT_BINARY_DIR}/bridge/target
-)
+
+if(APPLE AND CMAKE_OSX_ARCHITECTURES)
+
+    add_custom_target(cargo_mozillavpnnp_all COMMENT "Build vpnnp for all Architectures")
+    # For each requested Arch , create a new target and 
+    # make it a dependency for cargo_mozillavpnnp_all
+    foreach(OSXARCH ${CMAKE_OSX_ARCHITECTURES})
+        string(REPLACE "arm64" "aarch64" OSXARCH ${OSXARCH})
+        add_custom_target(cargo_mozillavpnnp_${OSXARCH}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bridge
```

Comment:
The indentation is a bit odd here, please indent `WORKING_DIRECTORY` and `COMMAND` by another step to make it clearer they are arguments to `add_custom_target()`